game-core/src/main/java/games/strategy/engine/framework/startup/ui/ClientSetupPanel.java

```diff
@@ -270,6 +269,11 @@ public IChatPanel getChatPanel() {
     return actions;
   }
 
+  @Override
+  public JComponent getDrawable() {
+    return this;
+  }
```

Comment:
In general I agree. In short the thinking was the instance of check was even more egregious. In specific with an interface (even with a bad abstraction) at least the contract is made explicit and we have opportunity to fail fast and detect our coding mistakes. So it's still bad, with luck it's an easier to fix and lesser degree of bad.

Would fixing this properly be reasonable for the scope of this branch? I'm happy to expand scope, but I think this might needs its own iteration/focus. The intended cleanup target for this branch is largely `MainPanel`.

Second question, do you feel strongly we should go back to the instanceof check? It's no comparison for me, instanceof check is much worse IMO, but maybe you do not share that point of view? I could be overlooking exactly how dirty this still is..

Reply:
I'm fine with changing this in the future.
I agree that instanceof is worse. What I'm saying is that the class implementing this interface should either not be the UI class itself, or the headless bot code should never be able to get its hands on an object that might be a UI instance.
Needs a lot of rethinking though, which would increase the scope of this PR by a lot.